lib/elixir/lib/string.ex

```diff
@@ -795,13 +794,21 @@ defmodule String do
     prefix_size = byte_size(string) - suffix_size
 
     case string do
-      <<prefix::size(prefix_size)-binary, suffix::size(suffix_size)-binary>> when suffix == match ->
-        prefix <> replacement
+      <<prefix::size(prefix_size)-binary, suffix::binary>> when suffix == match ->
+        append(prefix, replacement)
       _ ->
         string
     end
   end
 
+  @compile {:inline, prepend: 2, append: 2}
+
+  defp prepend("", suffix), do: suffix
+  defp prepend(prefix, suffix), do: prefix <> suffix
+
+  defp append(prefix, ""), do: prefix
+  defp append(prefix, suffix), do: prefix <> suffix
```

Comment:
@whatyouhide I like the specific functions, so I would go with `prepend_unless_empty`and `append_unless_empty`.